doc/src/base/arrays.md

```diff
@@ -115,6 +115,12 @@ Base.checkindex
 Base.elsize
 ```
 
+The following functions can be used to handle arrays with custom indices:
```

Comment:
I'm not particularly excited about publicizing `require_one_based_indexing` — the reason for its existence is entirely about adding safety checks to _old_ code.  But I see that the pair reference eachother and can be useful.

```suggestion
While most code can be written in an index-agnostic manner (see, e.g., [`eachindex`](@ref)), it can sometimes be useful to explicitly check for offset axes:
```

Reply:
I'd push back on this somewhat, as there are often operations (such as linear algebra ones) that don't have well-defined behaviour for offset arrays. In such cases, this function provides an easy way to restrict the method to a subset for which such an operation is sensible.